src/main/java/wolox/training/models/Book.java

```diff
@@ -124,63 +126,70 @@ public String getAuthor() {
     }
 
     public void setAuthor(String author) {
-        this.author = author;
+        this.author = Preconditions.checkNotNull(author,"Author must not be null");
     }
 
     public String getImage() {
         return image;
     }
 
     public void setImage(String image) {
-        this.image = image;
+
+        this.image = Preconditions.checkNotNull(image, "Image must not be null");
     }
 
     public String getTitle() {
         return title;
     }
 
     public void setTitle(String title) {
-        this.title = title;
+
+        this.title = Preconditions.checkNotNull(title, "Title must not be null");
     }
 
     public String getSubtitle() {
         return subtitle;
     }
 
     public void setSubtitle(String subtitle) {
-        this.subtitle = subtitle;
+
+        this.subtitle = Preconditions.checkNotNull(subtitle, "Subtitle must not be null");
     }
 
     public String getPublisher() {
         return publisher;
     }
 
     public void setPublisher(String publisher) {
-        this.publisher = publisher;
+
+        this.publisher = Preconditions.checkNotNull(publisher, "Publisher must not be null");
     }
 
     public String getYear() {
         return year;
     }
 
     public void setYear(String year) {
-        this.year = year;
+
+        this.year = Preconditions.checkNotNull(year,"Year must not be null");
     }
 
     public Integer getPages() {
         return pages;
     }
 
     public void setPages(Integer pages) {
-        this.pages = pages;
+
+        this.pages = Preconditions.checkNotNull(pages, "Pages must not be null");
```

Comment:
Another validation could be that the number of pages must be greater than zero :)